status.go

```diff
@@ -183,13 +182,12 @@ func statusLine(statusCode int) []byte {
 
 func invalidStatusLine(statusCode int) []byte {
 	statusText := StatusMessage(statusCode)
-	var buf bytes.Buffer
-	// xxx  placeholder of status code
-	buf.Grow(len("HTTP/1.1 xxx \r\n") + len(statusText))
-	buf.WriteString("HTTP/1.1 ")
-	buf.WriteString(strconv.Itoa(statusCode))
-	buf.WriteString(" ")
-	buf.WriteString(statusText)
-	buf.WriteString("\r\n")
-	return buf.Bytes()
+	// xxx placeholder of status code
+	var line = make([]byte, 0, len("HTTP/1.1 xxx \r\n")+len(statusText))
+	line = append(line, []byte("HTTP/1.1 ")...)
+	line = append(line, []byte(strconv.Itoa(statusCode))...)
```

Comment:
https://pkg.go.dev/strconv#AppendUint is even better here.

Reply:
yep 👍🏻    it also reduce memory allocs。 i will change to `AppendUint`  thx

Reply:
only  one time  memory allocation / op .  thanks  😊
AppendUint - benchmark 
```
goos: darwin
goarch: amd64
pkg: github.com/valyala/fasthttp
BenchmarkStatusLine512-8        48531439                33.2 ns/op            48 B/op          1 allocs/op
BenchmarkStatusLine512-8        52698516                22.7 ns/op            48 B/op          1 allocs/op
BenchmarkStatusLine512-8        52759131                22.9 ns/op            48 B/op          1 allocs/op
BenchmarkStatusLine512-8        52137784                23.2 ns/op            48 B/op          1 allocs/op
BenchmarkStatusLine512-8        53255018                24.3 ns/op            48 B/op          1 allocs/op
BenchmarkStatusLine512-8        50873232                23.7 ns/op            48 B/op          1 allocs/op
BenchmarkStatusLine512-8        51188386                23.7 ns/op            48 B/op          1 allocs/op
BenchmarkStatusLine512-8        42466096                24.5 ns/op            48 B/op          1 allocs/op
BenchmarkStatusLine512-8        40846321                25.7 ns/op            48 B/op          1 allocs/op
BenchmarkStatusLine512-8        41946541                29.5 ns/op            48 B/op          1 allocs/op
PASS
ok      github.com/valyala/fasthttp     24.393s
```

Reply:
i am  sorry.  if statusCode is negative,   [https://pkg.go.dev/strconv#AppendUint](https://pkg.go.dev/strconv#AppendUint)  will  cause incorrect result , as unit  test 
```
testStatusLine(t, -1, []byte("HTTP/1.1 -1 Unknown Status Code\r\n"))
```